src/parser/common.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <structures/structures.hpp>
+#include <lexer/lexer.hpp>
+#include <utils/utils.hpp>
+
+namespace lir::parser {
+
+    // Parser state passed throughout the parser internally
+    struct State {
+        Token previous;
+        Token current;
+
+        FileStack& files;
+
+        AST prefix_node;
+        AST infix_node;
+
```

Comment:
Mostly a personal preference thing but I think you should put `lir::` before those typenames.